server.go

```diff
@@ -510,22 +510,40 @@ func (s *Server) HandleTracePacket(packet []byte) {
 		return
 	}
 
-	sample, metrics, err := samplers.ParseSSF(packet)
+	msg, err := samplers.ParseSSF(packet)
 	if err != nil {
 		reason := fmt.Sprintf("reason:%s", err.Error())
 		s.Statsd.Count("packet.error_total", 1, []string{"packet_type:ssf_metric", reason}, 1.0)
 		log.WithError(err).Warn("ParseSSF")
 		return
 	}
-
+	metrics, err := msg.Metrics()
+	if err != nil {
+		reason := fmt.Sprintf("reason:%s", err.Error())
+		s.Statsd.Count("packet.error_total", 1, []string{
+			"packet_type:ssf_metric",
+			"step:extract_metrics",
+			reason,
+		}, 1.0)
+		log.WithError(err).Warn("ParseSSF-metrics")
+		if _, ok := err.(samplers.InvalidMetrics); !ok {
+			// This is a serious error, let's abort:
```

Comment:
Should we emit a separate metric here?

Reply:
Second thought: This already uses `reason` to distinguish between errors, so I think you can tell when something unexpected happens. I'll make the `reason` determination more meaningful, though.